packages/react-router-native/NativeRouter.js

```diff
@@ -23,67 +23,20 @@ class NativeRouter extends Component {
     }
   }
 
-  state = {
-    savedHistory: null
-  }
-
-  componentDidMount() {
-    AsyncStorage.getItem('history', (err, history) => {
-      this.setState({
-        savedHistory: history ? JSON.parse(history) : {
-          entries: this.props.initialEntries,
-          index: this.props.initialIndex
-        }
-      })
-    })
-  }
-
   render() {
-    const { getUserConfirmation, keyLength, children } = this.props
-    const { savedHistory } = this.state
+    const { initialEntries, initialIndex, getUserConfirmation, keyLength, children } = this.props
 
-    return savedHistory != null ? (
+    return (
       <MemoryRouter
-        initialEntries={savedHistory.entries}
-        initialIndex={savedHistory.index}
+        initialEntries={initialEntries}
+        initialIndex={initialIndex}
         getUserConfirmation={getUserConfirmation}
         keyLength={keyLength}
       >
-        <Route render={({ entries, index }) => (
-          <StoreHistory entries={entries} index={index}>
-            {React.Children.only(children)}
-          </StoreHistory>
-        )}/>
+        <Route render={() => React.Children.only(children)} />
```

Comment:
Just get that `<Route>` outta there and pass `children` straight on through to `<MemoryRouter>`. It will take care of rendering the `children` according to [the logic used by the rest of the routers](https://github.com/ReactTraining/react-router/blob/f86e4baab0abf2f67dfbe7d1b3b503f6ac16ab34/packages/react-router/modules/Router.js#L33-L34).